cbor.js

```diff
@@ -111,14 +195,18 @@ function encode(value) {
 
     switch (typeof value) {
       case "number":
+        if (isNaN(value) || !isFinite(value)) {
+          return writeFloat16(value);
+        }
+        
         if (Math.floor(value) === value) {
           if (0 <= value && value <= POW_2_53)
             return writeTypeAndLength(0, value);
           if (-POW_2_53 <= value && value < 0)
             return writeTypeAndLength(1, -(value + 1));
         }
-        writeUint8(0xfb);
-        return writeFloat64(value);
+
+        return writeFloat(value);
```

Comment:
floats are now encoded into 16, 32, and 64 bits where precision allows for exact transcoding